devito/core/gpu.py

```diff
@@ -156,19 +149,17 @@ def _specialize_clusters(cls, clusters, **kwargs):
         clusters = cire(clusters, 'invariants', sregistry, options, platform)
         clusters = Lift().process(clusters)
 
-        # Reduce flops (potential arithmetic alterations)
+        # Reduce flops
         clusters = extract_increments(clusters, sregistry)
         clusters = cire(clusters, 'sops', sregistry, options, platform)
         clusters = factorize(clusters)
         clusters = optimize_pows(clusters)
 
-        # Reduce flops (no arithmetic alterations)
-        clusters = cse(clusters, sregistry)
-
-        # Lifting may create fusion opportunities, which in turn may enable
-        # further optimizations
+        # The previous passes may have created fusion opportunities
         clusters = fuse(clusters)
-        clusters = eliminate_arrays(clusters)
+
+        # Reduce flops
+        clusters = cse(clusters, sregistry)
```

Comment:
Note for reviewers: note that now `cse` is performed after `fuse`. That's fine, it was wrong before